gossip/basiccheck_test.go

```diff
@@ -0,0 +1,913 @@
+			basiccheck.ErrZeroTime,
+		},
+		{
+			"Validate checkTxs validateTx ErrNegativeValue-1",
+			func() {
+				s.me.SetSeq(idx.Event(1))
+				s.me.SetEpoch(idx.Epoch(1))
+				s.me.SetFrame(idx.Frame(1))
+				s.me.SetLamport(idx.Lamport(1))
+
+				h := hash.BytesToEvent(bytes.Repeat([]byte{math.MaxUint8}, 32))
+				tx1 := types.NewTx(&types.LegacyTx{
+					Nonce:    math.MaxUint64,
+					GasPrice: h.Big(),
+					Gas:      math.MaxUint64,
+					To:       nil,
+					Value:    big.NewInt(-1000),
+					Data:     []byte{},
+					V:        big.NewInt(0xff),
+					R:        h.Big(),
+					S:        h.Big(),
+				})
+				txs := types.Transactions{}
+				txs = append(txs, tx1)
+				s.me.SetTxs(txs)
+			},
+			basiccheck.ErrNegativeValue,
+		},
+		{
+			"Validate checkTxs validateTx ErrNegativeValue-2",
+			func() {
+				s.me.SetSeq(idx.Event(1))
+				s.me.SetEpoch(idx.Epoch(1))
+				s.me.SetFrame(idx.Frame(1))
+				s.me.SetLamport(idx.Lamport(1))
+
+				h := hash.BytesToEvent(bytes.Repeat([]byte{math.MaxUint8}, 32))
+				tx1 := types.NewTx(&types.LegacyTx{
+					Nonce:    math.MaxUint64,
+					GasPrice: big.NewInt(-1000),
+					Gas:      math.MaxUint64,
+					To:       nil,
+					Value:    h.Big(),
+					Data:     []byte{},
+					V:        big.NewInt(0xff),
+					R:        h.Big(),
+					S:        h.Big(),
+				})
+				txs := types.Transactions{}
+				txs = append(txs, tx1)
+				s.me.SetTxs(txs)
+			},
+			basiccheck.ErrNegativeValue,
+		},
+		{
+			"Validate checkTxs validateTx ErrIntrinsicGas",
+			func() {
+				s.me.SetSeq(idx.Event(1))
+				s.me.SetEpoch(idx.Epoch(1))
+				s.me.SetFrame(idx.Frame(1))
+				s.me.SetLamport(idx.Lamport(1))
+
+				h := hash.BytesToEvent(bytes.Repeat([]byte{math.MaxUint8}, 32))
+				tx1 := types.NewTx(&types.LegacyTx{
+					Nonce:    math.MaxUint64,
+					GasPrice: h.Big(),
+					Gas:      0,
+					To:       nil,
+					Value:    h.Big(),
+					Data:     []byte{},
+					V:        big.NewInt(0xff),
+					R:        h.Big(),
+					S:        h.Big(),
+				})
+				txs := types.Transactions{}
+				txs = append(txs, tx1)
+				s.me.SetTxs(txs)
+			},
+			basiccheck.ErrIntrinsicGas,
+		},
+
+		/*
+			      {
+			         "Validate checkTxs validateTx ErrTipAboveFeeCap",
+			          func() {
+			            s.me.SetSeq(idx.Event(1))
+			            s.me.SetEpoch(idx.Epoch(1))
+			            s.me.SetFrame(idx.Frame(1))
+			            s.me.SetLamport(idx.Lamport(1))
+
+			            h := hash.BytesToEvent(bytes.Repeat([]byte{math.MaxUint8}, 32))
+
+			            tx1 := types.NewTx(&types.LegacyTx{
+			               Nonce:    math.MaxUint64,
+			               GasPrice: h.Big(),
+			               Gas:      math.MaxUint64,
+			               To:       nil,
+			               Value:    h.Big(),
+			               Data:     []byte{},
+			               V:        big.NewInt(0xff),
+			               R:        h.Big(),
+			               S:        h.Big(),
+			            })
+
+
+			            tx1 := types.NewTx(&types.DynamicFeeTx{
```

Comment:
I assume, `types.DynamicFeeTx` has to be used to test against [basiccheck.ErrTipAboveFeeCap](https://github.com/cyberbono3/go-opera/blob/basiccheck-testing/eventcheck/basiccheck/basic_check.go#L71) in this test case, has not it?  `types.LegacyTx` has no `GasFeeCap` field.

Reply:
Yes, that's right